### ISSUE TITLE
Vertex Module fails test out of the box

### ISSUE THREAD

**Issue by tylers-username** [magento/magento2#20827]
Magento 2.2.4

### Summary (*)
1. Setup fresh install of Magento 2.2.4
1. Follow Magento's instructions to prepare Unit tests
2. Execute: `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist --testsuite="Magento Unit Tests" vendor/vertex/module-tax/Test/Unit/Observer/CreditMemoObserverTest.php`

environments, and so on. -->

### Examples (*)
1. Passing tests

```
PHPUnit 6.2.4 by Sebastian Bergmann and contributors.

E                                                                   1 / 1 (100%)

Time: 1.26 seconds, Memory: 14.00MB

There was 1 error:

1) Vertex\Tax\Test\Unit\Observer\CreditMemoObserverTest::testSendRefundRequest
TypeError: Argument 1 passed to Mock_TaxInvoice_33ea0ed8::sendRefundRequest() must implement interface Vertex\Services\Invoice\RequestInterface, null given, called in /app/vendor/vertex/module-tax/Observer/CreditMemoObserver.php on line 106

/app/vendor/vertex/module-tax/Observer/CreditMemoObserver.php:106
/app/vendor/vertex/module-tax/Test/Unit/Observer/CreditMemoObserverTest.php:144

ERRORS!
Tests: 1, Assertions: 0, Errors: 1.

Process finished with exit code 2 at 20:27:45.
Execution time: 3,129 ms.
```
### Proposed solution

Additionally, I've seen in other issues related to Vertex in this repository be dismissed by the Magento team as "not their problem". However, I agree with others that by being included in the core product, Magento should support this module.

**Comment by magento-engcom-team** [comment]
Hi @tylerssn. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@tylerssn do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by tylers-username** [comment]
It turns out that vertex/module-tax does not have the appropriate composer dependency requirements. Version `vertex/module-tax:2.2.1` and up is not compatible with Magento 2.2.4's test suite. Downgrading to `vertex/module-tax:2.2.0` resolved the issue.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
Hi @navarr. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by navarr** [comment]
I suspect these issues to be fixed in the upcoming 2.3.1/2.2.8.

**Comment by tylers-username** [comment]
@navarr I see fixed in 2.2.x. Does this mean anything after 2.2.4?

**Comment by navarr** [comment]
This might still occur in 2.2.8+ as we were unable to release Vertex 3.1 with 2.2.8;

I've not tested it on 2.2.8

**Comment by engcom-Charlie** [comment]
Hello @tylerssn 

Thank you for contribution and collaboration!

We are not able to reproduce this issue on the lates 2.3-develop branch by provided steps.

We are closing this issue due to branch `2.2-develop` was closed and
> Magento no longer accepts pull requests and issues to the v2.2 release lines to focus all development efforts on v2.3.

 See [Accepted pull requests and ported code](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#rules) for more details

If you still faced this issue on `2.3` please create a new Issue with all required details according to [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines)

Thanks for your report!
